components/motor/gpio/encoded.go

```diff
@@ -411,11 +419,13 @@ func (m *EncodedMotor) IsMoving(ctx context.Context) (bool, error) {
 	return m.real.IsMoving(ctx)
 }
 
-// Stop stops rpmMonitor and stops the real motor.
+// Stop stops makeAdjustments and stops the real motor.
 func (m *EncodedMotor) Stop(ctx context.Context, extra map[string]interface{}) error {
-	if m.rpmMonitorDone != nil {
-		m.rpmMonitorDone()
+	m.mu.Lock()
```

Comment:
I'm assuming you're not using 
```go m.mu.Lock()
defer m.mu.Unlock()
```
because something hung somewhere?

Reply:
no this should be defer

Reply:
changed it in the latest commit